### PR TITLE
MED-171-fix-booking-UI

### PR DESCRIPTION
**Pull request by SoulOfDarknes**
1. Fixed pick slot time
2. fixed and make it more informative on meeting information
3. booking button dont active any more when you didnt choose a doctor 

[MED-102](https://github.com/ZenBit-Tech/MedOn_fe/pull/34)

**PR checklist:**

**1.1 Common**
- [x] 1.1.2 no any, should be also added to eslint rules
- [x] 1.1.1 types for input and output params
- [x] 1.1.3 try/catch for any async function
- [x] 1.1.4 no magic numbers
- [x] 1.1.5 compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] 1.1.6 no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] 1.1.7 avoid similar types with duplicating by generics
- [x] 1.1.8 no console.log in pr
- [x] 1.1.9 .env file should be in .gitignore
- [x] 1.1.10 delete commented code if it's not part of documentation
- [x] 1.1.11 no links in the code, env links should be in env file (for example: server url),
- [x] constant links (for example default avatar URL) should be in constant file
- [x] 1.1.12 constants and function names should be lowercase.
- [x] 1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] 1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’, const EACH_DAY_15h_15min = ‘15 3 * * *’

 **1.3 Frontend|Mobile**
- [x] 1.3.1 split component and logic (move logic to custom hook file)
- [x] 1.3.2 colors, font size, and font name should be in the theme or in the constants
- [x] 1.3.3 no text in the components, use i18n approach, even if you have only one language for now
- [x] 1.3.4 inline styles prohibited
- [x] 1.3.5 import should be absolute. instead of ../../../components/myComponent should be components/myComponent

[MED-102]: https://olhabilous.atlassian.net/browse/MED-102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/medon-fe/src/components/SelectDoctor/index.tsx [hunk 1]
```diff
@@ -41,6 +42,7 @@ export default function SelectDoctor({
   const [selectedSpeciality, setSelectedSpeciality] = useState<number | string>(
     'all'
   );
+  const [visibleDoctors, setVisibleDoctors] = useState<number>(3);
```
Comment (oleksii-yurchenko) [inline apps/medon-fe/src/components/SelectDoctor/index.tsx]: 3 move to consts

File: apps/medon-fe/src/components/SelectDoctor/index.tsx [hunk 2]
```diff
@@ -142,6 +154,11 @@ export default function SelectDoctor({
             </ItemWrap>
           </ListItem>
         ))}
+        {filteredDoctors.length > visibleDoctors && (
+          <LoadMore onClick={showMoreDoctors}>
+            {t('patient-list.load-more')}
```
Comment (oleksii-yurchenko) [inline apps/medon-fe/src/components/SelectDoctor/index.tsx]: why "patient-list" when we are working with doctors?

File: apps/medon-fe/src/components/SelectTimeSlot/index.tsx [hunk 3]
```diff
@@ -1,40 +1,53 @@
...
+        const isSlotAvailable = Object.values(timeSlotsAvailability).some(
+          (availability) => availability[index]
+        );
+
+        if (!isSlotAvailable) {
+          return null;
+        }
+
+        const doctorCount = Object.values(timeSlotsAvailability)
+          .map((availability) => availability[index])
```
Comment (oleksii-yurchenko) [inline apps/medon-fe/src/components/SelectTimeSlot/index.tsx]: instead using "map" and "filter" in general better to use "reduce". But in this case do we really need "map"?

File: apps/medon-fe/src/components/SelectDoctor/index.tsx (hunk 2 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/SelectDoctor/index.tsx]: so as not to duplicate the same text I took from another

File: apps/medon-fe/src/components/SelectDoctor/index.tsx (hunk 1 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/SelectDoctor/index.tsx]: done

File: apps/medon-fe/src/components/SelectTimeSlot/index.tsx (hunk 3 above)
Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/SelectTimeSlot/index.tsx]: In this particular case, the code uses the "map" and "filter" methods to perform two different tasks: obtaining physician availability values and filtering unavailable time slots. Using these methods in this context is logical and understandable.
"reduce" may complicate the code and make it less understandable.
But I change it to reduce :D

File: apps/medon-fe/src/components/SelectTimeSlot/index.tsx [hunk 4]
```diff
@@ -1,40 +1,53 @@
...
-              {
-                Object.values(timeSlotsAvailability).filter(
-                  (availability) => availability[index]
-                ).length
-              }
-            </span>
-          </DrText>
-        </TimeSlot>
-      ))}
+      {timeSlots.map((timeSlot, index) => {
```
Comment (lyudad) [inline apps/medon-fe/src/components/SelectTimeSlot/index.tsx:13]: double check that app will no fail if there no timeslots

Comment (SoulOfDarknes) [reply apps/medon-fe/src/components/SelectTimeSlot/index.tsx:13]: it does not fall, it just will not be available day or will not render the slot

**Review by skazzp — APPROVED** [review]

**Pull request merged**